sentinel.go

```diff
@@ -319,13 +338,29 @@ func (c *sentinelClient) listWatch(cc conn) (master string, sentinels []string,
 				if m[0] == "master" && m[1] == c.sOpt.Sentinel.MasterSet {
 					c.switchMasterRetry(fmt.Sprintf("%s:%s", m[2], m[3]))
 				}
+			case "+slave", "+sdown", "-sdown":
+				m := strings.SplitN(event.Message, " ", 4)
+				if m[0] == "slave" {
```

Comment:
We might also need to check `c.readonly` here as well.

Reply:
I think beside `c.readonly` we need to check `c.sOpt.Sentinel.MasterSet` in these events. because other master set events can trigger this client's refresh with current implementation.